jep/README.md

```diff
@@ -0,0 +1,207 @@
+
+This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your JEP with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+```
+
+The following constitutes a non-exclusive list of prior art:
+
+- JupyterLab extension example using a React.js component https://github.com/jupyterlab/extension-examples/tree/master/react-widget
+- Nteract is a React.js notebook https://github.com/nteract/nteract
+- Jupyter React is a library with React.js components to create data products compatible with the Jupyter ecosystem.https://github.com/datalayer/jupyter-react
+- [Jupyter Widgets](https://github.com/jupyter-widgets) for a reactive solution that imply both frontend and backend.
+- JupyterLab RTC for collaboration.
+- JupyterHub management UI on top of React.js Bootstrap widgets https://github.com/jupyterhub/jupyterhub/tree/main/jsx
+- [CoCalc](https://github.com/sagemathinc/cocalc) connects to Jupyter Kernels with a frontend developed in Reacts.js.
+- UI Toolkit is a component library for building web interfaces in Jupyter ecosystem (JupyterHub, Jupyter Widgets, JupyterLab,...) https://github.com/jupyterlab-contrib/jupyter-ui-toolkit. This JEP proposes to built on top tof this implementation.
+
+# Unresolved questions
+
+```
+- What parts of the design do you expect to resolve through the JEP process before this gets merged?
+- What related issues do you consider out of scope for this JEP that could be addressed in the future independently of the solution that comes out of this JEP?
+```
+
+A base widget toolkit is in-scope of this JEP. Reactivity and collaborative are out-of-scope of this JEP.
+
+It should be possible for a React.js widget to share a global state with a widget developed in another technology (Vue.js, Svelte...). This can be achieved by a adequate global state/store system that is not restricted a single technology (so should be `polyglot`). This can be defined in subsequent JEP.
+
+Reactivity to support features like [Jupyter Widgets](https://github.com/jupyter-widgets) can be defined in subsequent JEP. The current implementation of Jupyter Widgets relies on [Backbone.js](https://backbonejs.org) for which a successor would be welcome. We could imagine relying on the collaborative feature based on CRDT (Conflict-free replicated data types).
+
+Realtime collaboration and local-first aplication are not in-scope for now but it is logical to mention that and depend on a CRDT solution (Y.js in use for now).
+
+Integration with the current Lumino extension system could be considered in the future. As this is a bit more complex to design, especially to support multiple technologies (not only React.js), we think a separated discussion will be better for that.
```

Comment:
I think the integration with the existing Lumino framework and especially its extension system should be developed more.

One of the strengths of the JupyterLab ecosystem is its extension system. Folks will expect being able to reuse all the existing extensions that have been developed over the past years as is. So the UI toolkit presented here should somehow be compatible with it.

Reply:
Done with https://github.com/datalayer/jupyter-react/commit/eb97229989d1d2ad33fa9f26cfcc1ce988549849 Please resolve if it is clear enough.